Binary files bwa/QSufSort.o and bwa-mod/QSufSort.o differ
Binary files bwa/bamlite.o and bwa-mod/bamlite.o differ
Binary files bwa/bntseq.o and bwa-mod/bntseq.o differ
Binary files bwa/bwa and bwa-mod/bwa differ
Binary files bwa/bwa.o and bwa-mod/bwa.o differ
diff -rupN -x Xcode -x .git bwa/bwamem.c bwa-mod/bwamem.c
--- bwa/bwamem.c	2015-07-08 13:14:28.000000000 -0400
+++ bwa-mod/bwamem.c	2015-08-28 15:12:12.000000000 -0400
@@ -942,6 +942,39 @@ void mem_aln2sam(const mem_opt_t *opt, c
 		for (i = tmp; i < str->l; ++i) // replace TAB in the comment to SPACE
 			if (str->s[i] == '\t') str->s[i] = ' ';
 	}
+	if ((p->flag & 4) == 0) {
+		kputsn("\tZT:Z:", 6, str);
+		//kputs((char*)(p->cigar + p->n_cigar), str);
+		//kputc(',', str);
+		kputw(p->score, str); // same as AS:i
+		kputc(',', str);
+		if(p->sub >= 0) {
+			kputw(p->score - p->sub, str);
+		} else {
+			kputsn("NA", 2, str);
+		}
+		kputc(',', str);
+		kputw(p->pair_score, str);
+		kputc(',', str);
+		if(p->pair_sub >= 0) {
+			//assert(p->pair_score >= p->pair_sub);
+			kputw(p->pair_score - p->pair_sub, str);
+		} else {
+			kputsn("NA", 2, str);
+		}
+		kputc(',', str);
+		kputw(p->sub_n, str);
+		kputc(',', str);
+		kputw(p->pair_nsub, str);
+		kputc(',', str);
+		kputw((int)(p->frac_rep * 1000.0), str);
+		kputc(',', str);
+		kputw((int)(p->pair_frac_rep * 1000.0), str);
+		kputc(',', str);
+		kputw((int)(p->seedcov * 1000.0), str);
+		kputc(',', str);
+		kputw((int)(p->pair_seedcov * 1000.0), str);
+	}
 	kputc('\n', str);
 }
 
@@ -1131,6 +1164,11 @@ mem_aln_t mem_reg2aln(const mem_opt_t *o
 	a.pos = pos - bns->anns[a.rid].offset;
 	a.score = ar->score; a.sub = ar->sub > ar->csub? ar->sub : ar->csub;
 	a.is_alt = ar->is_alt; a.alt_sc = ar->alt_sc;
+	a.seedcov = ((float)ar->seedcov) / l_query;
+	a.frac_rep = ar->frac_rep;
+	a.sub_n = ar->sub_n;
+	a.pair_score = a.pair_sub = a.pair_nsub = 0;
+	a.pair_frac_rep = a.pair_seedcov = 0.;
 	free(query);
 	return a;
 }
diff -rupN -x Xcode -x .git bwa/bwamem.h bwa-mod/bwamem.h
--- bwa/bwamem.h	2015-07-08 13:14:28.000000000 -0400
+++ bwa-mod/bwamem.h	2015-08-06 16:18:00.000000000 -0400
@@ -94,6 +94,9 @@ typedef struct { // This struct is only 
 	char *XA;        // alternative mappings
 
 	int score, sub, alt_sc;
+	float seedcov, pair_seedcov;
+	int sub_n, pair_score, pair_sub, pair_nsub;
+	float frac_rep, pair_frac_rep;
 } mem_aln_t;
 
 #ifdef __cplusplus
Binary files bwa/bwamem.o and bwa-mod/bwamem.o differ
Binary files bwa/bwamem_extra.o and bwa-mod/bwamem_extra.o differ
diff -rupN -x Xcode -x .git bwa/bwamem_pair.c bwa-mod/bwamem_pair.c
--- bwa/bwamem_pair.c	2015-07-08 13:14:28.000000000 -0400
+++ bwa-mod/bwamem_pair.c	2015-08-06 16:18:28.000000000 -0400
@@ -346,6 +346,21 @@ int mem_sam_pe(const mem_opt_t *opt, con
 				aa[i][n_aa[i]++] = g[i];
 			}
 		}
+		// Might be less work if we installed these earlier
+		h[0].pair_score = h[1].pair_score = o;
+		h[0].pair_sub = h[1].pair_sub = subo;
+		h[0].pair_nsub = h[1].pair_nsub = n_sub;
+		h[0].pair_frac_rep = h[1].pair_frac_rep = a[0].a[0].frac_rep + a[1].a[0].frac_rep;
+		h[0].pair_seedcov = h[1].pair_seedcov = ((float)(a[0].a[0].seedcov + a[1].a[0].seedcov)) / (s[0].l_seq + s[1].l_seq);
+		for (j = 0; j < 2; ++j) {
+			for (i = 0; i < n_aa[j]; ++i) {
+				aa[j][i].pair_score = h[j].pair_score;
+				aa[j][i].pair_sub = h[j].pair_sub;
+				aa[j][i].pair_nsub = h[j].pair_nsub;
+				aa[j][i].pair_frac_rep = h[j].pair_frac_rep;
+				aa[j][i].pair_seedcov = h[j].pair_seedcov;
+			}
+		}
 		for (i = 0; i < n_aa[0]; ++i)
 			mem_aln2sam(opt, bns, &str, &s[0], n_aa[0], aa[0], i, &h[1]); // write read1 hits
 		s[0].sam = strdup(str.s); str.l = 0;
Binary files bwa/bwamem_pair.o and bwa-mod/bwamem_pair.o differ
Binary files bwa/bwape.o and bwa-mod/bwape.o differ
Binary files bwa/bwase.o and bwa-mod/bwase.o differ
Binary files bwa/bwaseqio.o and bwa-mod/bwaseqio.o differ
Binary files bwa/bwashm.o and bwa-mod/bwashm.o differ
Binary files bwa/bwt.o and bwa-mod/bwt.o differ
Binary files bwa/bwt_gen.o and bwa-mod/bwt_gen.o differ
Binary files bwa/bwt_lite.o and bwa-mod/bwt_lite.o differ
Binary files bwa/bwtaln.o and bwa-mod/bwtaln.o differ
Binary files bwa/bwtgap.o and bwa-mod/bwtgap.o differ
Binary files bwa/bwtindex.o and bwa-mod/bwtindex.o differ
Binary files bwa/bwtsw2_aux.o and bwa-mod/bwtsw2_aux.o differ
Binary files bwa/bwtsw2_chain.o and bwa-mod/bwtsw2_chain.o differ
Binary files bwa/bwtsw2_core.o and bwa-mod/bwtsw2_core.o differ
Binary files bwa/bwtsw2_main.o and bwa-mod/bwtsw2_main.o differ
Binary files bwa/bwtsw2_pair.o and bwa-mod/bwtsw2_pair.o differ
Binary files bwa/fastmap.o and bwa-mod/fastmap.o differ
Binary files bwa/is.o and bwa-mod/is.o differ
Binary files bwa/kopen.o and bwa-mod/kopen.o differ
Binary files bwa/kstring.o and bwa-mod/kstring.o differ
Binary files bwa/ksw.o and bwa-mod/ksw.o differ
Binary files bwa/kthread.o and bwa-mod/kthread.o differ
Binary files bwa/libbwa.a and bwa-mod/libbwa.a differ
Binary files bwa/main.o and bwa-mod/main.o differ
Binary files bwa/malloc_wrap.o and bwa-mod/malloc_wrap.o differ
Binary files bwa/maxk.o and bwa-mod/maxk.o differ
Binary files bwa/pemerge.o and bwa-mod/pemerge.o differ
Binary files bwa/utils.o and bwa-mod/utils.o differ
